### ISSUE TITLE
[Bug] Case insensitive image imports

### ISSUE THREAD

**Issue by davidedelvento** [ankidroid/Anki-Android#7973]
###### Reproduction Steps

1. I created a deck on desktop app (I did it on Linux, FWIW), containing two cards, each which one image, which were loaded from filesystem and were named respecively `F#m.png` and `F#M.png` (the images are similar, but different, as the name implies)
2. The deck on the desktop app works fine, showing different images for the different cards
3. Export the deck as single `.apkg` for only that deck, without scheduling information and **with** media
4. Send that file via email to myself, open in on AnkiDroid by clicking the email attachment
5. AnkiDroid imports the deck

Note: I never have edited the deck on AnkiDroid itself, nor changed anything about it (basically all the additions and changes are performed on the desktop app and one-way sent to the mobile device).

###### Expected Result

The deck should work fine on the AnkiDroid app too, showing different images for the different cards

###### Actual Result

The deck contains the same images for the different cards. Thinking of a one-off glitch, I deleted both cards, sent to myself another copy of the deck (to which I had added additional cards). The problem persisted, however the image shown on both cards now is the other one.

###### Debug info
AnkiDroid Version = 2.14.2
Android Version = 10
ACRA UUID = 6c891926-54bf-4a44-811f-959a6b440f29
Scheduler = std
Crash Reports Enabled = true

I suspect this is related to issue #6622
I also suspect this might be due to the original filename differing only by case (even though Linux fully distinguish them, and I suspect Android does too).

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison-1** [comment]
Thanks for logging the issue (and searching)!

We should be able to get this fixed, but there will likely always be issues with filenames that differ only by case (as we support syncing to Windows, which is case-preserving).

**Comment by davidedelvento** [comment]
Thank you for the extremely quick answer. Is the following a viable workaround which should not cause any obvious side effect (as far as you can tell, of course the unexpected is always possible, I know)?

1. I delete the affected cards both on AnkiDroid and on the desktop app
2. rename the files on the filesystem 
3. repeat steps above

I'll be eager to try that, but I don't want to risk (too much) my production learning experience with Anki and AnkiDroid (awesome project, btw)

**Comment by david-allison-1** [comment]
Sorry for the brevity, I'm about to sleep, you'll likely get a better answer on our [discord](https://discord.gg/qjzcRTx), or on [reddit](https://www.reddit.com/r/anki) 

----

To be honest, I don't see why you wouldn't use AnkiWeb to sync, so you likely won't hit the issue with imports.

----

If that isn't reasonable, a simple fix would be to rename the file to remove the ambiguity (`F#maj.jpg`)

You can then edit the cards, rather than delete and recreate them. If you're at a scale of hundreds of cards, hit the links above, the community is awesome.

**Comment by davidedelvento** [comment]
No worries and thanks again for the quick answer. I'm about to sleep too, I will try the workaround tomorrow and report back here whether or not it works (for others to possibly find and answer when looking).

Should the issue be not fixable (you mentioned compatibility with Windows, which I haven't used since 1999...) I suggest trying to detect the problem (or possible problem) and issue a warning when appropriate.

**Comment by davidedelvento** [comment]
So yes, the delete and recreate workaround works fine. I have not tested the modify workaround because in my case I learned all the wrong things and I definitely wanted all the wrong cards wiped out from my decks and not preserved with "wrong" due dates, leeches etc. I suspect that modifying would work fine too. I also preferred to do the work by hand since my decks are highly curated, and I had only a couple dozen cards to deal with, and I feared automation could have created additional problems.

Information for people landing on this issue: the problematic cards can be identified in AnkiDroid by using the card browser, because the file name is modified by appending a numeric string which happened to be identical for both the files which were supposed to be different by case. Hence just glancing at the card browser will immediately show these cards with "weird" filenames (assuming your normal filenames are not weird already :-) )

So it seems to me that AnkiDroid "knows" about this problem happening (since it adds a numeric suffix to the filename) just that the it acts incorrectly to "fix" it. I strongly recommend adding a warning immediately (in the next push release on all devices) if a proper fix takes more time. Corrupting a deck which one is learning is a **major** issue in my opinion which should be avoided at all cost, and if it can't be avoided at least a warning should be issued.

That said, thanks again for developing and maintaining such a great piece of software!

**Comment by davidedelvento** [comment]
Let me know if you need me to do anything for this.

**Comment by mikehardy** [comment]
The best path forward is:

> If the issue is still here, please keep in mind that we need community support and help to fix it!

...check the development guide and see if / what import is doing with image files that differ only in case to see why this is happening, and propose a change that resolves whatever the root cause is

**Comment by davidedelvento** [comment]
Why has this been closed?

I understand people having other priorities, and this bug happens only in an edge case, but it still is a serious data-corruption issue!

Closing it without a fix seem a bit extreme to me.

**Comment by mikehardy** [comment]
Because there are 292 open issues still. And there were thousands before. Please propose a fix, we'll happily merge it, but we can't just have an endless issues list. This is a community of people working together and the issues list should reflect actual ongoing work.

It's not like it is deleted or anything, this will always be here for reference if someone wants to take it up. In the meantime, your definition of serious apparently does not meet the threshold of anyone wanting to do anything. That is not intended to sound harsh - it is just objective reality

**Comment by davidedelvento** [comment]
Mike,

If you are dealing with thousands of open and stale issues, probably you don't have the time to dig deeper into all of them, so perhaps you have superficially reviewed this one too. Let me summarize it for you.

In some circumstances, images are **SILENTLY overwritten**. This will cause cards to display **WRONG** images and the user unknowingly will **learn the WRONG thing**. For an application like Anki, which prouds itself to make learning better, this is utmost unacceptable, in my opinion. Now, the "in some circumstances" part make this bug not an everyday occurrence for everybody, hence users are not up in arms and contributors are not rushing to provide a fix, but it is still extremely serious bug in my book. To be more clear, if I had not discovered this bug, I'd have learn something **WRONG** because of Anki!!!!!

Let me make a comparison: would you use a tax reporting software which in most cases works just fine but has a chance (however small) to submit random numbers to the IRS??

Also, your last point does not sound harsh, it sounds moot: nobody who wants to contribute some code to the project is going to look among thousands of closed bugs and find this one.

So in conclusion, I suspect in the very few time you had available for this issue, you overlooked its seriousness and hence treated it as a minor thing (for which closing if nobody is working on it at the moment would have been fine). I hope you will reconsider.

**Comment by davidedelvento** [comment]
Closing it you certainly will not get any PR. You should have added a "help wanted" label instead.

I am really baffled at your attitude: closing serious bugs, hoping to get PR on closed issues... what's next?

**Comment by mikehardy** [comment]
@davidedelvento I in turn am baffled at anyone who thinks lengthy issues with no proposals of code change help move anything forward in any serious way on a volunteer project? I'm happy to add tags though, FWIW
